docs/layers.md

```diff
@@ -2108,6 +2108,90 @@ Transit lines may have their colours mapped onto one of these CSS colours. The i
 * `yellow`
 * `yellowgreen`
 
+## Traffic Flow
+
+This **optional** layer is meant to be updated every few minutes, due to the highly dynamic nature of the data. Consideration should be given to rendering the features from this layer linked to the **roads** layer when linear referencing is available, or as an overlay
+
+* Layer name: `traffic_flow`
+* Geometry types: `line`
+
+#### `common` properties:
+
+* `id`: Unique traffic event ID. Can be referenced when checking for updated traffic information for specified event
+* `kind`: the **severity** of the flow information as seen by traffic provider to indicate the traffic "color"
+* `min_zoom`: this value is derived from combination of `road_kind`, `kind`, and `kind_detail` values. A suggestion for which zoom to draw a feature. The value is a float
+
+#### Possible `kind` values for traffic flow:
+
+* `unknown`: traffic status unknown
+* `free`: Free flowing, not disturbed, traffic
+* `minor`: Minor traffic
+* `slow`: Slowly moving traffic
+* `queuing`: The traffic is in queues but still moves slowly
+* `stationary`: Stationary traffic, congestion
+* `none`: No traffic flow due to blockage or closure
+
+#### `common-optional` properties:
+
+* `source`: provider of traffic information
+* `congestion`: the level of traffic flow, with 0.0 representing completely free flowing traffic to 1.0 completely congested traffic
+* `speed`: speed in km/h
+* `drives_on_left`: set to `true` when the country drives on the left, e.g. In the U.K
+
+#### `optional` properties:
+
+The following properties should only be included when **roads** layer features doesn't include `linear_ref_id`: identifier link to a linear references system, eg [SharedStreet](https://github.com/sharedstreets/sharedstreets-ref-system/blob/master/OSMLR.md). In that case the traffic data should be delivered without geometry and run-time linked with the **roads** layer. But baring that, the following `optional` properties enable sizing and layering of the traffic overlay with respect to the roads network:
```

Comment:
You say that the following properties should only be included when the **roads** layer features don't include `linear_ref_id`, but then the first list item is `linear_ref_id` (although it says "or.." at the end).

This confused me. Do you think it might be clearer if split into two lists, e.g:

> If the **roads** layer features include `linear_ref_id`, then traffic flow layer features may include the following, in which case the traffic data should be delivered without geometry and run-time linked with the **roads** layer.
>
> * `linear_ref_id`: identifier link to a linear references system, eg [SharedStreet](https://github.com/sharedstreets/sharedstreets-ref-system/blob/master/OSMLR.md)
>
> However, if the **roads** layer does not include `linear_ref_id`, then it should be delivered with geometry and the following properties to enable sizing and layering of the traffic overlay with respect to the roads network:
>
> * `road_kind`: [kind of the road](https://github.com/tilezen/vector-datasource/blob/master/docs/layers.md#road-properties-common)
> ...etc...

?

Reply:
Split these out :)